packages/ra-ui-materialui/src/input/SelectArrayInput.stories.tsx

```diff
@@ -0,0 +1,124 @@
+    DialogActions,
+    DialogContent,
+    TextField,
+} from '@mui/material';
+
+import { AdminContext } from '../AdminContext';
+import { Create } from '../detail';
+import { SimpleForm } from '../form';
+import { SelectArrayInput } from './SelectArrayInput';
+import { useCreateSuggestionContext } from './useSupportCreateSuggestion';
+
+export default { title: 'ra-ui-materialui/input/SelectArrayInput' };
+
+const i18nProvider = polyglotI18nProvider(() => englishMessages);
+
+export const Basic = () => (
+    <AdminContext i18nProvider={i18nProvider}>
+        <Create
+            resource="users"
+            record={{ roles: ['u001', 'u003'] }}
+            sx={{ width: 600 }}
+        >
+            <SimpleForm>
+                <SelectArrayInput
+                    source="roles"
+                    choices={[
+                        { id: 'admin', name: 'Admin' },
+                        { id: 'u001', name: 'Editor' },
+                        { id: 'u002', name: 'Moderator' },
+                        { id: 'u003', name: 'Reviewer' },
+                    ]}
+                    sx={{ width: 300 }}
+                />
+            </SimpleForm>
+        </Create>
+    </AdminContext>
+);
+
+const choices = [
+    { id: 'admin', name: 'Admin' },
+    { id: 'u001', name: 'Editor' },
+    { id: 'u002', name: 'Moderator' },
+    { id: 'u003', name: 'Reviewer' },
+];
+
+const CreateRole = () => {
+    const { onCancel, onCreate } = useCreateSuggestionContext();
+    const [value, setValue] = React.useState('');
+
+    const handleSubmit = event => {
+        event.preventDefault();
+        const newOption = { id: value, name: value };
+        choices.push(newOption);
+        setValue('');
+        onCreate(newOption);
+    };
+
+    return (
+        <Dialog open onClose={onCancel}>
+            <form onSubmit={handleSubmit}>
+                <DialogContent>
+                    <TextField
+                        label="Role name"
+                        value={value}
+                        onChange={event => setValue(event.target.value)}
+                        autoFocus
+                    />
+                </DialogContent>
+                <DialogActions>
+                    <Button type="submit">Save</Button>
+                    <Button onClick={onCancel}>Cancel</Button>
+                </DialogActions>
+            </form>
+        </Dialog>
+    );
+};
+
+export const CreateProp = () => (
+    <AdminContext i18nProvider={i18nProvider}>
+        <Create
+            resource="users"
+            record={{ roles: ['u001', 'u003'] }}
+            sx={{ width: 600 }}
+        >
+            <SimpleForm>
+                <SelectArrayInput
+                    source="roles"
+                    choices={choices}
+                    sx={{ width: 300 }}
+                    create={<CreateRole />}
+                />
+            </SimpleForm>
+        </Create>
+    </AdminContext>
+);
+
+export const EmptyText = () => (
```

Comment:
The name of this story is not related to what is showing, plus I don't get what is trying to show, it looks the same as the Basic story